source/common/http/codec_client.h

```diff
@@ -173,8 +179,15 @@ class CodecClient : Logger::Loggable<Logger::Id::client>,
     CodecReadFilter(CodecClient& parent) : parent_(parent) {}
 
     // Network::ReadFilter
-    Network::FilterStatus onData(Buffer::Instance& data, bool) override {
+    Network::FilterStatus onData(Buffer::Instance& data, bool end_stream) override {
       parent_.onData(data);
+      // Reset the codec client on end stream when HTTP1.
+      if (end_stream && parent_.isHalfCloseEnabled() && parent_.protocol() <= Protocol::Http11) {
```

Comment:
I'd be inclined to take out the protocol check here, since half close is only enabled for HTTP/1 (but if we added it for HTTP/2 we'd probably want to treat as full close)

Reply:
yes, I agree on this.